src/Npgsql/NpgsqlTypes/NpgsqlRange.cs

```diff
@@ -131,49 +199,215 @@ internal NpgsqlRange([CanBeNull] T lowerBound, [CanBeNull] T upperBound, RangeFl
                x.LowerBoundInfinite == y.LowerBoundInfinite &&
                x.UpperBoundInfinite == y.UpperBoundInfinite;
 
+        /// <summary>
+        /// Indicates whether the <see cref="NpgsqlRange{T}"/> on the left is not equal to the <see cref="NpgsqlRange{T}"/> on the right.
+        /// </summary>
+        /// <param name="x">The <see cref="NpgsqlRange{T}"/> on the left.</param>
+        /// <param name="y">The <see cref="NpgsqlRange{T}"/> on the right.</param>
+        /// <returns>
+        /// True if the <see cref="NpgsqlRange{T}"/> on the left is not equal to the <see cref="NpgsqlRange{T}"/> on the right; otherwise, false.
+        /// </returns>
         public static bool operator !=(NpgsqlRange<T> x, NpgsqlRange<T> y) => !(x == y);
 
-        public override bool Equals([CanBeNull] object o) => o is NpgsqlRange<T> && this == (NpgsqlRange<T>)o;
+        /// <inheritdoc />
+        public override bool Equals(object o) => o is NpgsqlRange<T> range && this == range;
 
+        /// <inheritdoc />
         public bool Equals(NpgsqlRange<T> other) => this == other;
 
+        /// <inheritdoc />
         public override int GetHashCode()
-            => IsEmpty ? 0 :
-                (LowerBoundInfinite ? 0 : LowerBound.GetHashCode()) +
-                (UpperBoundInfinite ? 0 : UpperBound.GetHashCode());
+            => IsEmpty
+                ? 0
+                : (LowerBoundInfinite ? 0 : LowerBound.GetHashCode()) +
+                  (UpperBoundInfinite ? 0 : UpperBound.GetHashCode());
 
+        /// <inheritdoc />
         public override string ToString()
         {
             if (IsEmpty)
                 return "empty";
+
             var sb = new StringBuilder();
-            if (LowerBoundInfinite)
-                sb.Append('(');
-            else
-            {
-                sb.Append(LowerBoundIsInclusive ? '[' : '(');
+
+            sb.Append(LowerBoundIsInclusive ? '[' : '(');
```

Comment:
Any reason for making the distinction between inclusive/exclusive when the bound is infinite? The previous code just outputted `(` for infinity...

Reply:
This change was motivated by the [PostgreSQL docs](https://www.postgresql.org/docs/current/static/rangetypes.html#RANGETYPES-INFINITE):

> there is no such thing as an inclusive infinite bound — if you try to write one, it will automatically be converted to an exclusive bound.

Since the constructor rewrite converts inclusive-infinity to exclusive-infinity, the distinction here is no longer needed. The `string` representation of the lower bound for any initialized `NpgsqlRange<T>` will only depend on the inclusivity test, because all infinite bounds will be constructed as exclusive.